### PR TITLE
feat: cms integration

### PR DESCRIPTION
**Pull request by alter-eggo**
Need to fix user-positions component here

### CODE REVIEW

**Comment by markmhendrickson** [comment]
@kyranjamie @camerow this has our combined changes for review

**Review by camerow — CHANGES_REQUESTED** [review]
This is a pretty tangled mess of types right now. I've pushed a change or two for eslint fixes and a few simpler type fixes, but based on the amount of typecasting happening I believe we really need to normalize the data and get the types nailed down. This is the most glaring issue with this PR, we should not see `any` / `unknown` or even really `as Foo` types needed since the data is so simple. Those type casts basically untype/retype the data. 

It's an easy today, hard tomorrow kind of solution.

**Comment by kyranjamie** [comment]
It's unnecessary to create new PRs for the same work as it gets rebased, @alter-eggo could have taken over from the previous PR, retaining all the feedback/comments.

**Comment by markmhendrickson** [comment]
In case helpful, @kyranjamie @camerow below is the refactor plan that I drafted and executed in response to your review, capturing the standards that were raised towards the end (which I'll load into my IDE as a rule set for further changes).

## Original Review Comments Status
✓ = Planned thoroughly for resolution
× = Not yet fully planned
[P1] = Resolved in Phase 1: Package Structure
[P2] = Resolved in Phase 2: Remove Value Defaults
[P3] = Resolved in Phase 3: Type System
[P4] = Resolved in Phase 4: Content Sanitization
[P5] = Resolved in Phase 5: Code Quality
[P6] = Resolved in Phase 6: Style System
[P7] = Resolved in Phase 7: Component Architecture
[P8] = Resolved in Phase 8: Convention Enforcement

### Linter Comments
- [✓][P5] post-info-hover-icon.tsx:25: "Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the 'void' operator"

### @kyranjamie Comments

#### Package Structure (4)
1. [✓][P1] root/package.json: "These deps should be added to the project where they're used, not the root"
2. [✓][P1] root/package.json: "also should be in web app not root" (prebuild script)
3. [✓][P1] packages/tsconfig-config/tsconfig.base.json: "This shouldn't be added to the base config" (Jest types)
4. [✓][P4] README.md:62-80: "Not sure this description belongs in the root readme, given it lives in the utils package"

#### Type System & Validation (8)
5. [✓][P3] pool-overview.tsx:22,43,69,91: "Dangerous type casting" (multiple instances)
6. [✓][P3] basic-hover-card.tsx:12-13: "Why is this casted to string, when it already is a string?"
7. [✓][P3] post-types.ts:25: "This should use `zod`"
8. [✓][P3] post-types.ts:16,25: "not using array[] - should use proper TypeScript array syntax"
9. [✓][P3] pool-overview.tsx:77: "Schema validation should be used instead of type casting"
10. [✓][P3] pool-overview.tsx:222,242,262,282,302: "Not clear why we use `React.JSX.IntrinsicElements`"
11. [✓][P7] apps/web/app/features/stacking/**/*.tsx: "Lots of repetition — could pass the PostCollection into the component"
12. [✓][P4] post-value-hover-card.tsx:14-29: "Should sanitize and style consistently"

#### Component Props & Defaults (10)
13. [✓][P2] pool-overview.tsx:21: "Default prop values should be avoided"
14. [✓][P2] pool-overview.tsx:21: "Default token to STX: risky default"
15. [✓][P2] post-label-hover-card.tsx:9-14: "Untyped props"
16. [✓][P2] pool-overview.tsx:19: "textStyle prop: not strongly typed to literal tokens → breaks contract"
17. [✓][P2] pool-overview.tsx:multiple: "Untyped textStyle"
18. [✓][P2] protocol-overview.tsx:108,111,171,191: "Multiple default props set: historicalApr, lockupPeriod, etc."
19. [✓][P6] value-displayer.tsx:10: "These string literals are redundant — likely available in FlexProps"
20. [✓][P6] post-section-heading.tsx:28: "whiteSpace: 'pre-line' logic should be centralized"
21. [✓][P2] post-label-hover-card.tsx:9-14: "Props are untyped"
22. [✓][P2] pool-overview.tsx:21,42,111: "Unnecessary default values"

#### Code Quality (4)
23. [✓][P5] pool-overview.tsx:82: "Pointless fragment use"
24. [✓][P5] post-page-heading.tsx:15-20: "Use of OR logic (||) instead of nullish coalescing (??)"
25. [✓][P5] post-label-hover-card.tsx:EOF: "Missing newline at end of file"
26. [✓][P5] post-page-heading.tsx:15: "Random comment"

#### Unused Imports/Variables (8)
27. [✓][P8] apy-hero-card.tsx:1: "'styled' is defined but never used"
28. [✓][P8] explainer.tsx:5: "'PostValueHoverCard' is defined but never used"
29. [✓][P8] explainer.tsx:6: "'PostLabelHoverCard' is defined but never used"
30. [✓][P8] pool-overview.tsx:6: "'PoolRewardProtocolInfo' is defined but never used"
31. [✓][P8] pool-overview.tsx:8: "'PostValueHoverCard' is defined but never used"
32. [✓][P8] pool-overview.tsx:11: "'ProviderIcon' is defined but never used"
33. [✓][P8] pool-overview.tsx:6,8,11: "~/data/data import is duplicated"
34. [✓][P8] apps/web/app/**/*.{ts,tsx}: "Multiple unused imports across components"

#### Style System (8)
- [✓][P6] post-info-hover-icon.tsx:36: "Not using panda styled.a"
- [✓][P6] post-info-hover-icon.tsx:40: "should be panda style props" (regarding inline styles)
- [✓][P6] value-displayer/default-value-displayer.tsx:10: "Panda probably has these string literals so we don't have to use them ourselves. Also, these are already in FlexProps so completely redundant to add again"
- [✓][P6] post-section-heading.tsx:28: "whiteSpace: 'pre-line' logic should be centralized"
- [✓][P6] post-info-hover-icon.tsx:multiple: "Should be using Panda-style props"
- [✓][P6] multiple files: "Use Panda's layout components (Flex, Box, etc.)"
- [✓][P6] multiple files: "Use Panda's spacing tokens"
- [✓][P6] multiple files: "Use Panda's color tokens"

#### Function Declarations (2)
- [✓][P8] post-info-hover-icon.tsx:23-26: "breach of convention [Function Declarations](https://www.notion.so/trustmachines/Function-declarations-22f7055df3124d868a242adb04f0f60d)"
- [✓][P8] multiple files: "We should follow a consistent manner for defining components"

### @camerow Comments
35. [✓][P3] apps/web/app/data/content.ts:22,77 apps/web/app/features/stacking/**/*.tsx:multiple: "This is a pretty tangled mess of types right now... we should not see any / unknown or even really as Foo types needed since the data is so simple. Those type casts basically untype/retype the data."

## Implementation Plan

### 1. Package Structure & Dependencies
**Resolves Comments:** # 1, # 2, # 3

#### Changes Required
1. Move dependencies from root to specific packages:
```json
// Move from root/package.json to apps/web/package.json
{
  "dependencies": {
    "@stacks/stacking": "7.0.5",
    "bignumber.js": "9.1.2",
    "dompurify": "3.2.4",
    "radix-ui": "1.3.4",
    "react-router-dom": "7.6.0"
  }
}
```

2. Move build script:
```json
// Move from root/package.json to apps/web/package.json
{
  "scripts": {
    "prebuild": "node scripts/fetch-remote-posts.cjs"
  }
}
```

#### Files to Update
- root/package.json
- apps/web/package.json
- packages/ui/package.json

### 2. Remove Value Defaults
**Resolves Comments:** # 13, # 14, # 15, # 16, # 17, # 18, # 21, # 22

#### Changes Required
1. Remove monetary/token defaults:
```typescript
// Before
function TotalValueLockedCell({
  totalValueLocked = '51,784,293 STX',
  totalValueLockedUsd = '$36,212,756'
}: Props)

// After
interface TotalValueLockedCellProps {
  totalValueLocked: string;    // Required
  totalValueLockedUsd: string; // Required
}
```

2. Update all components using defaults:
- MinimumCommitmentCell
- RewardTokenCell
- TotalValueLockedCell

#### Files to Update
```
apps/web/app/components/pool-overview.tsx
apps/web/app/features/stacking/**/*.tsx
```

### 3. Type System & Validation
**Resolves Comments:** # 5, # 6, # 7, # 8, # 9, # 10, # 35

#### A. Post Slug Normalization
```typescript
// Add utility
function toCamelCase(str: string): string {
  return str.replace(/-([a-z])/g, g => g[1].toUpperCase());
}

// Update normalization
export function normalizePosts(rawPosts: Record<string, any>): PostsCollection {
  const normalized: PostsCollection = {};
  for (const slug in rawPosts) {
    const camelCaseKey = toCamelCase(slug);
    normalized[camelCaseKey] = normalizePost(rawPosts[slug]);
  }
  return normalized;
}
```

#### B. Type Validation
```typescript
// Before
const posts = content.posts as unknown as PostsCollection;

// After
const postsSchema = z.record(z.string(), postSchema);
const posts = postsSchema.parse(content.posts);

// Type-safe access
function getPost(key: string): Post | undefined {
  return posts[key];
}
```

#### C. Array Type Syntax
Current Issue:
```typescript
// ❌ Incorrect: Using Array<T> syntax
views: string[];
earnProviders: any[];
icon: Array<{ name: string; url: string }>;
images: Array<{ name: string; url: string }>;
```

Solution:
```typescript
// ✅ Correct: Using T[] syntax
views: string[];
earnProviders: EarnProvider[]; // With proper type instead of any
icon: ImageAsset[];  // Extract repeated type
images: ImageAsset[];

// Define shared types
interface ImageAsset {
  name: string;
  url: string;
}
```

Rules:
1. Always use T[] syntax over Array<T>
2. Extract repeated types to interfaces
3. Avoid any[] in favor of specific types
4. Use consistent array syntax across codebase

Files to Update:
```
apps/web/app/data/post-types.ts:14-25
[any files using Array<T> syntax]
```

#### D. JSX Element Types (New Section)
Addresses comment # 10: "Not clear why we use `React.JSX.IntrinsicElements`"

Current Usage Found:
1. pool-overview.tsx (lines 222,242,262,282,302):
```typescript
tagName?: keyof React.JSX.IntrinsicElements;
```

2. post-label-hover-card.tsx (line 12):
```typescript
tagName?: string; // Untyped version
```

3. stacking-form-item-title.tsx (line 10):
```typescript
labelTagName?: 'h1' | 'h2' | 'h3' | 'h4' | 'span'; // Good pattern to follow
```

Problem:
- Using JSX.IntrinsicElements is overly permissive
- Inconsistent typing across components (string vs union vs JSX.IntrinsicElements)
- Some components using untyped string
- Makes component behavior less predictable

Solution:
1. Define specific allowed elements in shared types:
```typescript
// In shared/types.ts
export type HeadingTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
export type TextTag = 'p' | 'span' | 'div';
export type LabelTag = 'label' | 'span';
export type TextElementTag = HeadingTag | TextTag | LabelTag;

// Usage in components
interface Props {
  tagName?: TextElementTag;
}
```

2. Add validation helper:
```typescript
export const isValidTextElementTag = (tag: string): tag is TextElementTag => {
  const validTags = [
    'h1', 'h2', 'h3', 'h4', 'h5', 'h6',
    'p', 'span', 'div',
    'label'
  ];
  return validTags.includes(tag);
};
```

Files to Update:
```
apps/web/app/shared/types.ts (new file)
apps/web/app/components/pool-overview.tsx:222,242,262,282,302
apps/web/app/components/post-label-hover-card.tsx:12
[any other files using tagName props]
```

Testing:
- Test invalid tag rejection
- Test valid tag acceptance
- Test type inference
- Test component rendering with different tags
- Test validation helper
- Test each component variant

#### Files to Update
```
apps/web/app/data/content.ts
apps/web/app/data/post-types.ts
apps/web/app/utils/post-utils.ts
apps/web/app/components/**/*.tsx
```
#### E. React Element Type Standardization (New Section)
Addresses comment from @kyranjamie about using ReactNode instead of JSX.Element

Current Usage Found:
```typescript
// In choose-liquid-stacking-conditions.tsx (original)
const iconMap: Record<string, JSX.Element> = {
  // Icons here
};

// After PR updates to ReactNode
const iconMap: Record<string, ReactNode> = {
  BoxedCatLockedIcon: <BoxedCatLockedIcon />,
  MagnifyingGlassIcon: <MagnifyingGlassIcon />,
  StacksIcon: <StacksIcon />,
};
```
Problems:
- JSX.Element is too specific and restrictive
- Inconsistent type usage across the codebase (JSX.Element vs ReactNode vs ReactElement)
- Using JSX.Element prevents passing string, number, or null as valid React children

Solution:
1. Standardize on ReactNode for component children and general React content:
```typescript
// For component children and general content that could include strings, numbers, etc.
type SomeComponentProps = {
  children: ReactNode;
  content: ReactNode;
};
```
2. Use ReactElement when specifically requiring rendered components:
```typescript 
// For rendered React elements only (not strings/numbers)
type IconMapType = Record<string, ReactElement>;
```
3. Document standard usage in a shared types file:
```typescript
// In shared/types.ts
/**
 * Use ReactNode for:
 * - Component children
 * - Content that might include strings, numbers, or elements
 * - Any prop that accepts mixed content types
 */

/**
 * Use ReactElement for:
 * - When you need specifically a rendered React element
 * - Icon collections
 * - Element registries
 */
```
Files to Update:
```
apps/web/app/shared/types.ts
apps/web/app/features/stacking/**/*.tsx
[Any files using JSX.Element]
```
Testing:
- Validate type compatibility
- Ensure components accept appropriate children types
- Test edge cases (null, undefined, fragments)
- Verify consistent usage across codebase

### 4. Content Sanitization
**Resolves Comments:** # 4, # 12

#### Changes Required
1. Move sanitization to UI package:
```typescript
// Move from utils to UI package
import { sanitizeContent } from '@leather.io/ui/utils/sanitize-content';
```

2. Update documentation:
- Remove from root README
- Add to UI package docs
- Update all imports

#### Files to Update
```
packages/utils/src/sanitize-content.ts -> packages/ui/src/utils/sanitize-content/
apps/web/app/components/**/*.tsx
root/README.md
packages/ui/README.md
```

### 5. Code Quality & Conventions
**Resolves Comments:** # 23, # 24, # 25, # 26

#### A. Promise Handling (New Section)
Current Issue:
```typescript
// Unhandled promise in post-info-hover-icon.tsx:25
navigate(getPostHref(post.slug));
```

Solution:
```typescript
// Add proper error handling
void navigate(getPostHref(post.slug)).catch(error => {
  console.error('Navigation failed:', error);
  // Add error handling
});

// Or if navigation should block:
try {
  await navigate(getPostHref(post.slug));
} catch (error) {
  console.error('Navigation failed:', error);
  // Add error handling
}
```

ESLint Rule:
```json
{
  "rules": {
    "@typescript-eslint/no-floating-promises": "error"
  }
}
```

#### B. Function Declaration Convention

Rules:

1. **Hook Declarations**
```typescript
// ✅ Correct: Function declaration for hooks
export function useSomeHook(): SomeType {
  return {};
}

// ❌ Incorrect: Arrow function/const for hooks
export const useSomeHook = (): SomeType => {
  return {};
};
```

2. **Component Props Pattern**
```typescript
// ✅ Correct: Interface extension
interface SomeComponentProps extends BoxProps {
  tx: Transaction;
}

export function SomeComponent(props: SomeComponentProps) {
  return <>{/*...*/}</>;
}

// ❌ Incorrect: Type intersection and FC
export const SomeComponent: FC<{ tx: Transaction } & BoxProps> = () => {
  return <>{/*...*/}</>;
};
```

Key Rules:
1. Always use `function` declaration for hooks
2. Never use arrow functions for top-level declarations
3. Define props interface separately and use extension
4. Avoid type intersections with `&`
5. Never use `FC` type
6. Use explicit prop interfaces

ESLint Rules:
```json
{
  "rules": {
    "react/function-component-definition": ["error", {
      "namedComponents": "function-declaration",
      "unnamedComponents": "arrow-function"
    }],
    "@typescript-eslint/explicit-function-return-type": ["error", {
      "allowExpressions": true
    }],
    "@typescript-eslint/no-unused-vars": "error",
    "@typescript-eslint/consistent-type-definitions": ["error", "interface"]
  }
}
```

Files to Update:
```
apps/web/app/components/post-info-hover-icon.tsx:23-26
[all component and hook files]
```

Testing:
- Verify function declarations
- Check prop type inheritance
- Test component children handling
- Validate interface extensions

#### Changes Required
1. Remove unnecessary fragments:
```typescript
// Before
value={<>{minimumCommitment}</>}

// After
value={minimumCommitment}
```

2. Update logical operators:
```typescript
// Before
const label = post?.title || defaultTitle;

// After
const label = post?.title ?? defaultTitle;
```

3. Add ESLint rules:
- Enforce newlines at end of files
- Enforce function declaration style
- Enforce no unused imports

#### Files to Update
```
.eslintrc.js
apps/web/app/**/*.tsx
apps/web/app/**/*.ts
```

#### Code Organization (New Section)
- [✓][P5] stacking-form-item-title.tsx:17: "Why isn't this imported at top of file?"
- [✓][P5] multiple files: "Organize imports consistently"

### 6. Style System Improvements
**Resolves Comments:** # 19, # 20

#### A. Panda Props Usage
Current Issue:
```typescript
// ❌ Incorrect: Redundant prop types
export interface ValueDisplayerProps extends FlexProps {
  textAlign?: 'left' | 'right' | 'center'; // Already in FlexProps
}

// ❌ Incorrect: Custom string literals
type TextAlign = 'left' | 'right' | 'center';
```

Solution:
```typescript
// ✅ Correct: Use existing FlexProps types
export interface ValueDisplayerProps extends FlexProps {
  // No need to redeclare props that exist in FlexProps
  name: ReactNode;
  value: ReactNode;
}

// ✅ Correct: Use Panda's types
import { TextAlignProperty } from 'leather-styles/types';
```

Rules:
1. Don't redeclare props that exist in Panda's types
2. Use Panda's type definitions
3. Extend existing prop interfaces
4. Remove redundant type declarations

Files to Update:
```
apps/web/app/components/value-displayer/default-value-displayer.tsx:7-10
[any files with custom style prop types]
```

#### B. Centralize style tokens
```typescript
// Create shared tokens
export const styleTokens = {
  whiteSpace: {
    preLine: 'pre-line'
  }
} as const;
```

#### Files to Update
```
apps/web/app/styles/tokens.ts
```

### 7. Component Architecture
**Resolves Comments:** # 11

#### Current Pattern
```typescript
// Repeated in multiple components
const posts = content.posts as unknown as PostsCollection;
const label = posts['stacking-contract-details']?.title ?? 'Default';
```

#### Proposed Pattern
```typescript
interface PostLabelProps {
  posts: PostsCollection;  // Pass collection as prop
  postKey: string;
  defaultLabel?: string;
}

// Shared component
function PostLabel({ posts, postKey, defaultLabel }: PostLabelProps) {
  const label = posts[postKey]?.title ?? defaultLabel;
  return <PostLabelHoverCard postKey={postKey} label={label} />;
}

// Usage
function StackingContractDetailsLabel(props: StackingLabelProps) {
  return <PostLabel postKey="stacking-contract-details" defaultLabel="Details" {...props} />;
}
```

#### Files to Update
```
apps/web/app/components/pool-overview.tsx:200-253 (multiple label components)
apps/web/app/features/stacking/**/*.tsx (similar patterns)
```

#### Implementation Steps
1. Create shared PostLabel component
2. Pass PostsCollection through context or props
3. Update all label components to use shared component
4. Remove duplicate posts access code
5. Add proper typing and validation

#### Testing Requirements
1. Test PostLabel component
2. Test with missing posts
3. Test with missing labels
4. Test default fallbacks
5. Test type safety

### 8. Convention Enforcement
**Resolves Comments:** # 27, # 28, # 29, # 30, # 31, # 32, # 33, # 34

#### Changes Required
1. Add ESLint rules for conventions
2. Standardize function declarations
3. Enforce Panda style usage
4. Add import organization rules

#### Files to Update
```
.eslintrc.js
apps/web/app/**/*.tsx
apps/web/app/**/*.ts
```

## Testing Requirements

### Test Coverage Requirements
- Each changed file must maintain or improve coverage
- New utilities must have 100% coverage
- Changed components must have interaction tests
- Changed types must have validation tests

### Testing by Phase

#### 1. Package Structure
- Verify dependency resolution
- Test build process
- Validate import paths
- Test package boundaries

#### 2. Value Props
- Test required props enforcement
- Test prop type validation
- Test error states
- Test parent component updates

#### 3. Type Safety
- Test Zod schema validation
- Test type constraints
- Test error handling
- Test data transformations
- Test post slug normalization

#### 4. Content Sanitization
- Test sanitization in all contexts
- Test SSR sanitization
- Test error cases
- Test security boundaries

#### 5. Code Quality
- Test ESLint rules
- Test logical operators
- Test fragment removal
- Test code formatting

#### 6. Style System
- Test style token usage
- Test component variants
- Test responsive styles
- Test style composition

#### 7. Component Architecture
- Test component composition
- Test prop passing
- Test reusable patterns
- Test component integration

#### 8. Convention Enforcement
- Test ESLint rules
- Test style conventions
- Test import organization
- Test naming conventions

### Test Types Required
1. Unit Tests
   - Individual functions
   - Utility methods
   - Type validation
   - Data transformations

2. Integration Tests
   - Component composition
   - Data flow
   - Event handling
   - Error boundaries

3. E2E Tests
   - Critical paths
   - User workflows
   - Error scenarios
   - Performance impacts

4. Static Analysis
   - Type checking
   - Linting rules
   - Style validation
   - Import validation

### Testing Tools
- Jest/Vitest for unit tests
- React Testing Library for components
- Playwright for E2E
- TypeScript for static analysis
- ESLint for code quality
- Prettier for formatting

## Implementation Order

### Phase Dependencies
- Phase 1 (Package Structure) must complete before Phase 3 (Type System)
- Phase 2 (Value Defaults) should complete before Phase 7 (Component Architecture)
- Phase 3 (Type System) enables Phase 4 (Content Sanitization)
- Phase 8 (Convention Enforcement) depends on all other phases

### Recommended Sequence
1. **Phase 1: Package Structure & Dependencies**
   - Enables proper type imports
   - Sets up project structure
   - Required for subsequent phases

2. **Phase 2: Remove Value Defaults**
   - Removes dangerous defaults
   - Prepares for type system changes
   - Independent of package structure

3. **Phase 3: Type System**
   - Builds on clean package structure
   - Implements core type safety
   - Enables subsequent phases

4. **Phase 4: Content Sanitization**
   - Uses new type system
   - Implements content safety
   - Can run parallel with Phase 5

5. **Phase 5: Code Quality**
   - Remove unnecessary fragments
   - Update logical operators
   - Add ESLint rules
   - Fix unused imports
   
6. **Import Organization**
    - Move all imports to top of files
    - Group imports by type
    - Remove inline imports
    - Add import ordering rules
    - Update ESLint config

7. **Phase 6: Style System**
   - Builds on type system
   - Can run parallel with Phase 7
   - Improves component consistency

8. **Phase 7: Component Architecture**
   - Uses new type system
   - Builds on removed defaults
   - Improves component structure

9. **Phase 8: Convention Enforcement**
   - Final phase
   - Enforces all standards
   - Validates all changes

### Parallel Work Opportunities
- Phases 4 & 5 can run in parallel
- Phases 6 & 7 can run in parallel
- Documentation can be updated throughout

### Risk Mitigation
- Each phase has its own testing requirements
- Changes can be rolled back by phase
- Each phase produces valid intermediate state

## Omitted Sections
The following sections were deliberately omitted to keep the plan focused:
1. Component State Defaults (due to complexity)
2. Style System Defaults (to maintain focus on critical issues)

## Code Standards Summary

### Package Organization
1. Dependencies should live in the package where they're used (# 1, # 2)
2. Configuration should be package-specific, not in base configs (# 3)
3. Documentation should live with the code it documents (# 4)

### Type Safety
1. No type casting without validation (# 5, # 6, # 35)
2. Use Zod for runtime validation (# 7)
3. Use proper TypeScript array syntax (# 8)
4. Schema validation over type assertions (# 9)
5. Consistent component prop typing (# 15, # 21)
6. Use specific union types instead of broad type definitions (# 10)
7. Maintain consistent type patterns across similar components (# 10)
8. Use ReactNode for component children and general content
9. Use ReactElement when specifically requiring rendered components only
10. Use specific union types for HTML element names instead of JSX.IntrinsicElements

### Component Design
1. No default prop values that could mask errors (# 13, # 14, # 18, # 22)
2. Props should be strongly typed to available tokens (# 16, # 17)
3. Reuse existing style tokens and types (# 19, # 20)
4. Pass shared data through props rather than accessing globally (# 11)
5. Consistent sanitization patterns (# 12)
6. Use shared type definitions for common props (# 10)
7. Validate component props at runtime (# 10)

### Code Quality
1. No unnecessary fragments (# 23)
2. Use nullish coalescing (??) over OR (||) (# 24)
3. Files must end with newline (# 25)
4. No redundant comments (# 26)
5. No unused imports (# 27-# 34)

### Naming & Style Conventions
1. Use camelCase for all identifiers (# 1)
2. Follow Panda style system conventions (# 19, # 20)
3. Centralize shared styles (# 20)
4. Use proper component imports (# 27-# 34)

### Data Management
1. Validate data at boundaries (# 7, # 9)
2. Use proper typing for collections (# 35)
3. Avoid redundant data access (# 33)
4. Centralize shared data patterns (# 11)

### Best Practices
1. Follow internal conventions consistently
2. Keep code DRY - avoid repetition (# 11)
3. Use proper TypeScript features (# 7, # 8)
4. Maintain clean import structure (# 33, # 34)
5. Use framework features appropriately (# 19, # 20)
6. Prefer explicit unions over broad types (# 10)
7. Centralize common type definitions (# 10)
8. Add runtime validation for type safety (# 10)

### Error Handling
1. All Promises must be properly handled (#ESLint error)
2. Use try/catch for critical paths
3. Use void operator for intentionally ignored promises
4. Add error logging and recovery
5. Handle navigation promises appropriately

### Function Declarations
1. Use function declarations for hooks and components (#kyranjamie convention link)
2. Never use FC type or arrow functions for top-level declarations
3. Define props using interfaces, not type intersections
4. Use interface extension over type intersection
5. Use descriptive names over brevity
6. Follow consistent declaration patterns
7. Always declare props interface separately
8. Follow project function declaration conventions

### Naming Conventions (New Section)
1. Names should accurately describe what a variable is or what a function does
2. Prioritize meaning over brevity
3. Use camelCase for all identifiers
4. Follow consistent naming patterns across similar components

### Style System Standards
1. Use Panda styled components for all HTML elements
2. Use Panda style props instead of inline styles
3. Use Panda's design tokens for spacing, colors
4. Use Panda's layout components
5. No raw style objects or HTML elements
6. Consistent spacing and layout patterns

### Code Organization Standards
1. All imports at top of file
2. Imports grouped by type/source
3. No inline imports
4. Clear separation between imports
5. Consistent import ordering
6. Type imports separate from implementation

These standards are derived from @kyranjamie's and @camerow's review comments and represent the project's coding guidelines and best practices.

File: apps/web/app/components/post-value-hover-card.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  }
+  
+  if (!post) return null;
+  
+  return (
+    <Flex flexDir="column">
+      <PostInfoHoverIcon post={post}>
+        <styled.span textStyle="label.01">{sanitizeContent(label ?? post.title)}</styled.span>
+      </PostInfoHoverIcon>
+      {typeof value === 'string' ? (
```
Comment (kyranjamie) [inline apps/web/app/components/post-value-hover-card.tsx:29]:
```suggestion
      {isString(value) ? (
```
we have a helper

File: apps/web/app/data/posts.json.d.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+declare module './posts.json' {
+  import type { PostsCollection } from './post-types';
+  const value: PostsCollection;
+  export default value;
+}
```
Comment (kyranjamie) [inline apps/web/app/data/posts.json.d.ts:5]: I think this should not be needed anymore?

File: eslint.config.js [hunk 3]
```diff
@@ -50,6 +50,45 @@ export default tseslint.config(
...
+            'position': 'after'
+          }
+        ],
+        'newlines-between': 'always',
+        'alphabetize': {
+          'order': 'asc',
+          'caseInsensitive': true
+        }
+      }]
     },
```
Comment (kyranjamie) [inline eslint.config.js:92]: Why have these rules been added changed?

`'no-duplicate-imports': 'off',` we want this on

Comment (kyranjamie) [reply eslint.config.js:92]: We use prettier to enforce import ordering etc

Comment (kyranjamie) [reply eslint.config.js:92]: We also define rules in our eslint package, not directly in the config. Suggest removing these

File: apps/web/app/components/explainer.tsx [hunk 4]
```diff
@@ -69,8 +71,8 @@ export function ExplainerStep({ index, title, description, img }: EarnInstructio
           {title}
         </styled.h3>
       </VStack>
-      <styled.p color="ink.text-subdued" textStyle="caption.01" fontSize="13px">
-        {description}
+      <styled.p color="black" textStyle="caption.01" fontSize="13px">
+        {String(description || '')}
```
[the remark this answers is not in the document]
Comment (markmhendrickson) [reply apps/web/app/components/explainer.tsx:75]: Agreed, this was a stray from the last refactor. Removing now

File: apps/web/app/components/explainer.tsx [hunk 5]
```diff
@@ -69,7 +71,7 @@ export function ExplainerStep({ index, title, description, img }: EarnInstructio
           {title}
         </styled.h3>
       </VStack>
-      <styled.p color="ink.text-subdued" textStyle="caption.01" fontSize="13px">
+      <styled.p color="black" textStyle="caption.01" fontSize="13px">
```
Comment (kyranjamie) [inline apps/web/app/components/explainer.tsx:74]: this should use a color token fyi

File: apps/web/app/components/post-info-hover-icon.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  const navigate = useNavigate();
+  
+  if (!post) return <Fragment>{children}</Fragment>;
+  
+  const iconColorToken = iconColor === 'white' ? 'invert' : 'ink.text-subdued';
+  
+  const handleIconClick = (e: React.MouseEvent): void => {
+    e.stopPropagation();
+    void navigate(getPostHref(post.slug));
+  };
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply apps/web/app/components/post-info-hover-icon.tsx:33]: @markmhendrickson comment unaddressed

File: eslint.config.js (hunk 3 above)
Comment (camerow) [reply eslint.config.js]: @kyranjamie that was my bad, I was seeing complaints about duplicate imports from 
```js
import type MyType from 'foo';
import useThings from 'foo';
```

sounds like maybe it was my environment though

**Comment by markmhendrickson** [comment]
https://github.com/leather-io/mono/pull/1248
